atomate/vasp/firetasks/parse_outputs.py

```diff
@@ -775,6 +776,590 @@ def run_task(self, fw_spec):
+    """
+    Analyze the linear response data generated from get_wf_hubbard_hund_linresp to
+    compute Hubbard U (and Hund J) value(s).
+
+    Required parameters:
+        num_perturb (int): number of perturbed sites
+        spin_polarized (bool):
+        relax_nonmagnetic (bool):
+        db_file (str): path to the db file that holds your tasks
+            collection and that you want to hold the hubbard_hund_linresp
+            collection
+        wf_uuid (str): auto-generated from get_wf_hubbard_hund_linresp,
+            used to make it easier to retrieve task docs
+    """
+
+    required_params = ["num_perturb", "spin_polarized", "relax_nonmagnetic",
+                       "db_file", "wf_uuid"]
+    optional_params = []
+
+    summaries = []
+
+    def run_task(self, fw_spec):
+
+        uuid = self["wf_uuid"]
+        db_file = env_chk(self.get("db_file"), fw_spec)
+        to_db = self.get("to_db", True)
+
+        mmdb = VaspCalcDb.from_db_file(db_file, admin=True)
+
+        num_perturb_sites = int(self["num_perturb"])
+        spin_polarized = bool(self["spin_polarized"])
+        relax_nonmagnetic = bool(self["relax_nonmagnetic"])
+
+        keys = ['Ground state', 'NSCF', 'SCF']
+        response_dict = {'Ground state': {}, 'NSCF': {}, 'SCF': {}}
+        perturb_dict = {}
+
+        magnet_order_gs = None
+
+        for key in keys:
+
+            for i in range(num_perturb_sites):
+                response_dict[key].update({'Vup_site'+str(i): [],
+                                           'Nup_site'+str(i): []})
+                response_dict[key].update({'Vdn_site'+str(i): [],
+                                           'Ndn_site'+str(i): []})
+                response_dict[key].update({'Ntot_site'+str(i): [],
+                                           'Mz_site'+str(i): []})
+            response_dict[key].update({'magnetic order': []})
+
+        docs = list(mmdb.collection.find({"wf_meta.wf_uuid": uuid}))
+
+        # Find electron type responses for each site
+        inv_block_dict = {"0": "s", "1": "p", "2": "d", "3": "f"}
+        ldaul_vals = [-1 for i in range(num_perturb_sites)]
+        for i in range(num_perturb_sites):
+            for d in docs:
+                try:
+                    ldaul = int(d['calcs_reversed'][0]['input']['incar']['LDAUL'][i])
+                    if ldaul > ldaul_vals[i]:
+                        ldaul_vals[i] = ldaul
+                        break
+                except Exception as exc:
+                    logger.warning("Failed to obtain ldaul: ", exc)
+
+        for d in docs:
+
+            struct = Structure.from_dict(
+                d["calcs_reversed"][-1]["output"]['structure'])
+
+            incar_dict = d['calcs_reversed'][0]['input']['incar']
+            outcar_dict = d['calcs_reversed'][0]['output']['outcar']
+
+            # Check if task is used in LR analysis
+            use_calc = False
+            rkey = ""
+            if int(incar_dict["ICHARG"]) == 11:
```

Comment:
This line caused a KeyError on testing. Using `.get()` allows for the logic to work even if the key is not specified.

Also, a more compact way of writing lines 857:866 might be:

```
use_calc = (int(incar_dict.get("ICHARG") == 11) or (relax_nonmagnetic and int(incar_dict.get("ISPIN")==2)
```

(This isn't more or less right, just personal preference)

Reply:
I'll see what I can do. The reason I had it in that form was because I am setting `rkey` and `is_gs`, in addition to `use_calc`. I'll try to make the logic cleaner that ends at line 880. Thank you!

Reply:
I changed the logic at lines 861-866 to 

`use_calc = (not relax_nonmagnetic) or (relax_nonmagnetic and int(incar_dict.get("ISPIN"))==2)`